### DISCUSSION TITLE
How to run similar github action on local changes of private tap?

### DISCUSSION THREAD

**Discussion by ezchi** [Homebrew/discussions#1808] [Tap maintenance and Homebrew development]
Learn to create and maintain a private tap. Would like to know how could I run test-pot checking as done in github action with my local changes?

**Comment by SMillerDev** [comment]
You can run test-bot locally

**Comment by gromgit** [comment]
https://docs.brew.sh/Manpage#test-bot-options-formula

**Comment by ezchi** [comment]
I modified my formula at `~/repo/Formula/foo.rb`. 
Does it mean I have to modify the formula at `brew --repo user/repo` location to run command like `brew tap-test --only-tap-syntax user/repo/foo`?

**Reply by gromgit** [reply]
> I modified my formula at `~/repo/Formula/foo.rb`.
> Does it mean I have to modify the formula at `brew --repo user/repo` location to run command like `brew tap-test --only-tap-syntax user/repo/foo`?

1. If that expands to `/Users/ezchi/repo/Formula/foo.rb`, that's not where Homebrew expects to find a tap:
```bash
$ sed 's/class Hello/class HelloTest/' < $(brew --repo homebrew/core)/Formula/hello.rb > ~/tmp/hello-test.rb
$ brew test-bot ~/tmp/hello-test.rb
[...]
==> Running FormulaeDetect#detect_formulae!
    url               (blank)
    tap origin/master (blank)
    HEAD              (blank)
    diff_start_sha1   87b8b81032e99cdfe96f121a8c1b68c4a43f63e7
    diff_end_sha1     87b8b81032e99cdfe96f121a8c1b68c4a43f63e7

    testing_formulae  hello-test
    added_formulae    (none)
    modified_formulae (none)
    deleted_formulae  (none)
Error: No available formula with the name "hello-test".
```

So yes, you need to put `foo.rb` in a proper `user/repo` tap:
```bash
brew tap-new user/repo
mv ~/repo/Formula/foo.rb $(brew --repo user/repo)/Formula/
brew info user/repo/foo  # to double-check it's in the right place for Homebrew to find
```

2. I've never heard of `brew tap-test`, so I assume you mean `brew test-bot`.

3. As the documentation for `test-bot` that I linked to earlier says:

> * `--tap`: Use the Git repository of the given tap. Defaults to the core tap for syntax checking.
> [...]
> * `--only-tap-syntax`: Only run the tap syntax check step.

So `brew test-bot --only-tap-syntax user/repo/foo` only does syntax checks **in `homebrew/core`**, not `user/repo`:
```bash
$ brew test-bot --only-tap-syntax nonexistent/tap
==> Using Homebrew/homebrew-test-bot b588d39 (Merge pull request #645 from Homebrew/dawidd6-patch-1)
==> Using Homebrew/brew 3.2.1-70-g5659d74ff (Merge pull request #11646 from chenrui333/add-test-epub)
==> Using Homebrew/linuxbrew-core 87b8b81032e (gocryptfs: add 2.0.1 bottle.)

==> Running TapSyntax#run!
==> brew readall --aliases homebrew/core
==> brew style homebrew/core
==> brew audit --tap=homebrew/core --except=version
All steps passed!
```
